Dockerfile

```diff
@@ -91,14 +89,22 @@ RUN . $UNDERLAY_WS/install/setup.sh && \
     colcon build \
       --symlink-install \
       --mixin $OVERLAY_MIXINS \
-    || touch build_failed && \
-    if [ -f build_failed ] && [ -n "$FAIL_ON_BUILD_FAILURE" ]; then \
-      exit 1; \
-    fi
+    || ([ -z "$FAIL_ON_BUILD_FAILURE" ] || exit 1)
 
 # source overlay from entrypoint
 ENV UNDERLAY_WS $UNDERLAY_WS
 ENV OVERLAY_WS $OVERLAY_WS
 RUN sed --in-place \
       's|^source .*|source "$OVERLAY_WS/install/setup.bash"|' \
       /ros_entrypoint.sh
+
+# test overlay build
+ARG RUN_TESTS
```

Comment:
Should we be doing tests as part of the build for the development docker image?

Reply:
I left the `ARG` empty to disable testing during the dockerfile build by default, as there is no need for DockerHub to run tests when CI is already running them nightly. It's more to provide users a quick means to run tests locally while compiling a fresh build, or to self document how to build and test the project's workspace.